ml/src/main/scala/org/apache/spark/ml/h2o/models/H2OMOJOModel.scala

```diff
@@ -225,85 +218,23 @@ class H2OMOJOModel(val mojoData: Array[Byte], override val uid: String)
-    val qualifiedOutputPath = outputPath.makeQualified(fs.getUri, fs.getWorkingDirectory)
-    val out = fs.create(qualifiedOutputPath)
-    try {
-      out.write(instance.mojoData)
-    } finally {
-      out.close()
-    }
-    logInfo(s"Saved to: $qualifiedOutputPath")
-  }
-}
-
-private[models] class H2OMOJOModelReader[T <: H2OMOJOModel : ClassTag]
-(val defaultFileName: String) extends MLReader[T] {
-
-  @org.apache.spark.annotation.Since("1.6.0")
-  override def load(path: String): T = {
-    val metadata = DefaultParamsReader.loadMetadata(path, sc)
-
-    val inputPath = new Path(path, defaultFileName)
-    val fs = inputPath.getFileSystem(sc.hadoopConfiguration)
-    val qualifiedInputPath = inputPath.makeQualified(fs.getUri, fs.getWorkingDirectory)
-    val is = fs.open(qualifiedInputPath)
-
-    val mojoData = Stream.continually(is.read()).takeWhile(_ != -1).map(_.toByte).toArray
-
-    val h2oModel = make(mojoData, metadata.uid)(sqlContext)
-    metadata.getAndSetParams(h2oModel)
-    h2oModel
-  }
+object H2OMOJOModel extends H2OMOJOReadable[PyH2OMOJOModel] with H2OMOJOLoader[PyH2OMOJOModel] {
 
-  def make(mojoData: Array[Byte], uid: String)(sqLContext: SQLContext): T = {
-    classTag[T].runtimeClass.getConstructor(classOf[Array[Byte]], classOf[String]).
-      newInstance(mojoData, uid).asInstanceOf[T]
-  }
-}
-
-class H2OMOJOModelHelper[T <: py_sparkling.ml.models.H2OMOJOModel](implicit m: ClassTag[T]) extends MLReadable[T] {
-  val defaultFileName = "mojo_model"
-
-  @Since("1.6.0")
-  override def read: MLReader[T] = new H2OMOJOModelReader[T](defaultFileName)
-
-  @Since("1.6.0")
-  override def load(path: String): T = super.load(path)
-
-  def createFromMojo(path: String): T = {
-    val inputPath = new Path(path)
-    val fs = inputPath.getFileSystem(SparkSession.builder().getOrCreate().sparkContext.hadoopConfiguration)
-    val qualifiedInputPath = inputPath.makeQualified(fs.getUri, fs.getWorkingDirectory)
-    val is = fs.open(qualifiedInputPath)
-
-    createFromMojo(is, new File(path).getName)
-  }
-
-  def createFromMojo(is: InputStream, uid: String = Identifiable.randomUID("mojoModel")): T = {
-    val mojoData = Stream.continually(is.read).takeWhile(_ != -1).map(_.toByte).toArray
+  override def createFromMojo(mojoData: Array[Byte], uid: String): PyH2OMOJOModel = {
     val mojoModel = ModelSerializationSupport.getMojoModel(mojoData)
-    val sparkMojoModel = m.runtimeClass.getConstructor(classOf[Array[Byte]], classOf[String]).
-      newInstance(mojoData, uid).asInstanceOf[T]
+    val model = new PyH2OMOJOModel(uid)
+    model.setMojoData(mojoData)
     // Reconstruct state of Spark H2O MOJO transformer based on H2O's Mojo
     if (mojoModel.isSupervised) {
-      sparkMojoModel.setFeaturesCols(mojoModel.getNames.filter(_ != mojoModel.getResponseName))
-      sparkMojoModel.setLabelCol(mojoModel.getResponseName)
+      model.setFeaturesCols(mojoModel.getNames.filter(_ != mojoModel.getResponseName))
```

Comment:
What about filtering fold and weight column? Is it relevant?

Reply:
Good point -> we will solve this by using the MK change. Right now the values even though set, were not used anywhere